app/assets/javascripts/components/students/shared/StudentList/StudentDrawer/TrainingStatus/ExerciseRows.jsx

```diff
@@ -16,22 +16,46 @@ const ExerciseStatusCell = ({ status, sandboxUrl }) => {
 // Helper Functions
 const orderByDueDate = (a, b) => (moment(a.due_date).isBefore(b.due_date) ? -1 : 1);
 
-const generateRow = status => (exercise) => {
+// This function returns the status of exercise
+// returns status as unread/complete/incomplete depending on the deadline status and flags if any
+const getExerciseStatus = (exercise) => {
```

Comment:
I took help of [this](https://github.com/WikiEducationFoundation/WikiEduDashboard/blob/65fdeb29a79d122569db636f0d3d6faab9dff98e/app/assets/javascripts/reducers/exercises.js#L22) file to check status of exercise.